device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -421,158 +356,95 @@ private void closeAsync()
     }
 
-    private Integer sendMessage(AmqpsConvertToProtonReturnValue protonReturnValue, String deviceId) throws TransportException
-    {
-        Integer deliveryTag = -1;
-
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_015: [If the state of the connection is DISCONNECTED or there is not enough
-        // credit, the function shall return -1.]
-        if (this.state == IotHubConnectionStatus.DISCONNECTED || protonReturnValue == null)
-        {
-            log.trace("Amqp connection is disconnected, rejecting attempt to send message with delivery tag -1");
-            deliveryTag = -1;
-        }
-        else
-        {
-            Message protonMessage = protonReturnValue.getMessageImpl();
-            MessageType messageType = protonReturnValue.getMessageType();
-            // Codes_SRS_AMQPSIOTHUBCONNECTION_12_024: [The function shall call AmqpsSessionManager.sendMessage with the given parameters.]
-            deliveryTag = this.amqpsSessionManager.sendMessage(protonMessage, messageType, deviceId);
-        }
-
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_021: [The function shall return the delivery hash.]
-        return deliveryTag;
-    }
-
-    /**
-     * Event handler for reactor init event.
-     *
-     * @param event Proton Event object
-     */
     @Override
     public void onReactorInit(Event event)
     {
         Reactor reactor = event.getReactor();
 
-        // Codes_SRS_AMQPSIOTHUBCONNECTION_15_033: [The event handler shall set the current handler to handle the connection events.]
-        if (this.useWebSockets)
+        String hostName = this.hostName;
+        int port = AMQP_PORT;
+
+        if (this.deviceClientConfig.isUseWebsocket())
         {
             ProxySettings proxySettings = this.deviceClientConfig.getProxySettings();
             if (proxySettings != null)
             {
-                reactor.connectionToHost(proxySettings.getHostname(), proxySettings.getPort(), this);
+                hostName = proxySettings.getHostname();
+                port = proxySettings.getPort();
             }
             else
             {
-                reactor.connectionToHost(this.chooseHostname(), AMQP_WEB_SOCKET_PORT, this);
+                port = WEB_SOCKET_PORT;
             }
         }
-        else
-        {
-            reactor.connectionToHost(this.chooseHostname(), AMQP_PORT, this);
-        }
 
+        reactor.connectionToHost(hostName, port, this);
         reactor.schedule(SEND_MESSAGES_PERIOD_MILLIS, this);
 
         if (this.deviceClientConfig.getAuthenticationProvider() instanceof IotHubSasTokenAuthenticationProvider)
         {
             int sasTokenRenewalPeriod = this.deviceClientConfig.getSasTokenAuthentication().getMillisecondsBeforeProactiveRenewal();
-            reactor.schedule(sasTokenRenewalPeriod, sasTokenRenewalHandler);
+            reactor.schedule(sasTokenRenewalPeriod, new AmqpSasTokenRenewalHandler(this, this.deviceClientConfig));
```

Comment:
what is the behavior here for scenarios where user supplied the sas token directly (which is not auto-renewable)?